apps/hash-graph/docker/Dockerfile

```diff
@@ -20,14 +20,20 @@ RUN apk add --no-cache make openssl-dev protobuf-dev
 RUN --mount=type=cache,target=/usr/local/cargo/registry \
     --mount=type=cache,target=/usr/local/cargo/git \
     --mount=type=cache,target=/usr/local/src/hash-graph/target \
-    cargo install --path bin/cli --profile $PROFILE --locked
+    cargo install --path bin/cli --profile $PROFILE --features type-fetcher --locked
```

Comment:
Having this here would currently crash the production instance as it would spin up a server without a type-fetcher. Ideally we would have this be a built time arg in the dockerfile such that we can condtionally build this feature in until we fix the CD

Reply:
We discussed offline that we will merge the binaries (done in #2037) and #2038 adds a conditional feature flag instead of hardcode it